### PR TITLE
Replaced deprecated API endpoints and added channel list paging

### PR DESCRIPTION
**Pull request by shamarashid**
Replaced deprecated channels endpoints with conversations endpoints. Slack announcement  https://api.slack.com/changelog/2020-01-deprecating-antecedents-to-the-conversations-api#:~:text=The%20final%20date%20for%20existing,All%20channels. 

Added ability to iterate over large list of channels using paging and cursors.

### CODE REVIEW

File: slack_autoarchive.py [hunk 1]
```diff
@@ -78,16 +78,28 @@ def slack_api_http(
...
+                    self.logger.error(
+                        'Need to setup auth. eg, SLACK_TOKEN=<secret token> python slack-autoarchive.py'
+                    )
+                    sys.exit(1)
+                elif response.json()['error'] == 'not_in_channel':
+                    self.logger.error(
+                        'Need to add bot to channel described by ' + str(payload)
+                    )
+                    self.join_channel(payload['channel'])
+                    return response.json()
```
Comment (nilsreichardt) [inline slack_autoarchive.py]:
```suggestion
                    return self.slack_api_http(api_endpoint, payload, method)
```

TL;DR: Without this fix, the bot will archive **EVERY** channel

If you just return the `response` the following will happen:
1. Bot tries to read history -> rejected
2. Bot joins channel
3. Return `repsonse`, but in `response` is the message "not_in_channel"
4. Because no messages are in the request -> bot will archive the channel

**Review by nilsreichardt — CHANGES_REQUESTED** [review]

File: slack_autoarchive.py (hunk 1 above)
Comment (shamarashid) [reply slack_autoarchive.py]: @nilsreichardt : Thanks for catching the bug. Suggested changes applied.
